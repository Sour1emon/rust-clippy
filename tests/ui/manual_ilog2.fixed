#![warn(clippy::manual_ilog2)]

fn main() {
    let a: u32 = 5;
    let _ = a.ilog2();
    let _ = a.ilog2();

    let b: u64 = 543534;
    let _ = b.ilog2();

    let _ = 64 - b.leading_zeros(); // No lint
}
