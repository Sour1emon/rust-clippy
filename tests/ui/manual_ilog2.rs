#![warn(clippy::manual_ilog2)]

fn main() {
    let a: u32 = 5;
    let _ = 31 - a.leading_zeros();
    let _ = a.ilog(2);

    let b: u64 = 543534;
    let _ = 63 - b.leading_zeros();

    let _ = 64 - b.leading_zeros(); // No lint because manual ilog2 is BIT_WIDTH - 1 - x.leading_zeros()
}
