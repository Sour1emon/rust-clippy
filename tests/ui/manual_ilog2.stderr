error: manually reimplementing `ilog2`
  --> tests/ui/manual_ilog2.rs:5:13
   |
LL |     let _ = 31 - a.leading_zeros();
   |             ^^^^^^^^^^^^^^^^^^^^^^ help: consider using .ilog2(): `a.ilog2()`
   |
   = note: `-D clippy::manual-ilog2` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::manual_ilog2)]`

error: manually reimplementing `ilog2`
  --> tests/ui/manual_ilog2.rs:6:13
   |
LL |     let _ = a.ilog(2);
   |             ^^^^^^^^^ help: consider using .ilog2(): `a.ilog2()`

error: manually reimplementing `ilog2`
  --> tests/ui/manual_ilog2.rs:9:13
   |
LL |     let _ = 63 - b.leading_zeros();
   |             ^^^^^^^^^^^^^^^^^^^^^^ help: consider using .ilog2(): `b.ilog2()`

error: aborting due to 3 previous errors

